subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/downstream/test/TestDownstreamPlugin.java

```diff
@@ -22,12 +27,30 @@
 
     public void apply(final Project project) {
         IncubatingWarning.warn("downstream-testing plugin");
-        TaskMaker.task(project, TEST_DOWNSTREAM_TASK, TestDownstreamTask.class, new Action<TestDownstreamTask>() {
+
+        project.getPlugins().apply(UploadGistsPlugin.class);
+
+        TestDownstreamTask testDownstreamTask = TaskMaker.task(project, TEST_DOWNSTREAM_TASK, TestDownstreamTask.class, new Action<TestDownstreamTask>() {
             @Override
             public void execute(TestDownstreamTask task) {
                 task.setDescription("Runs all downstream tests.");
             }
         });
+
+        final UploadGistsTask uploadGistsTask = (UploadGistsTask) project.getTasks().findByName(UploadGistsPlugin.UPLOAD_GISTS_TASK);
+
+        uploadGistsTask.setFilesToUpload(
+            project
+                .fileTree(testDownstreamTask.getLogDirectory())
```

Comment:
Check the API of fileTree. It has some nice syntax for matching and including specific files. It will make this code more elegant. I don't remember the API but you probably can do something like:

```
fileTree(testDownstreamTask.getLogDirectory()).include("**/*.log")
```